linera-execution/src/execution_state_actor.rs

```diff
@@ -402,12 +403,25 @@ where
 
             ReadBlobContent { blob_id, callback } => {
                 let blob = self.system.read_blob_content(blob_id).await?;
+                if blob_id.blob_type == BlobType::Data {
+                    resource_controller
+                        .with_state(&mut self.system)
+                        .await?
+                        .track_blob_read(blob.bytes().len() as u64)?;
+                }
```

Comment:
There should be some constant cost for trying to read a blob even when it doesn't exist - here, if the blob is not found we'll return early and not track any reading-related cost.

Reply:
But then we fail the whole block anyway.

Reply:
Ah, I wanted to say it should be paid for up until that point in execution but we don't have invalid blocks.

Reply:
That's a discussion that goes well beyond the scope of this PR. Currently failing operations make the block invalid.